coderd/users.go

```diff
@@ -198,27 +198,32 @@ func (api *API) users(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	users, err := api.Database.GetUsers(ctx, database.GetUsersParams{
+	userRows, err := api.Database.GetUsers(ctx, database.GetUsersParams{
 		AfterID:   paginationParams.AfterID,
 		OffsetOpt: int32(paginationParams.Offset),
 		LimitOpt:  int32(paginationParams.Limit),
 		Search:    params.Search,
 		Status:    params.Status,
 		RbacRole:  params.RbacRole,
 	})
-	if errors.Is(err, sql.ErrNoRows) {
-		httpapi.Write(ctx, rw, http.StatusOK, []codersdk.User{})
-		return
-	}
 	if err != nil {
 		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
 			Message: "Internal error fetching users.",
 			Detail:  err.Error(),
 		})
 		return
 	}
+	// GetUsers does not return ErrNoRows because it uses a window function to get the count.
+	// So we need to check if the userRows is empty and return an empty array if so.
+	if len(userRows) == 0 {
+		httpapi.Write(ctx, rw, http.StatusOK, codersdk.GetUsersResponse{
+			Users: []codersdk.User{},
+			Count: 0,
+		})
+		return
+	}
```

Comment:
I don't think it ever could return that error anyways since it returns an array, and only `:one` queries will error. I think this simplifies the code below though, which is nice.